docs/enhancements/sep-tarball-client.md

```diff
@@ -0,0 +1,189 @@
+}
+
+// Host contains the metadata for a given host in a Kubernetes cluster.
+type Host struct {
+    // Name is the name of the host.
+    Name string
+    // Configz is the result of the configz endpoint.
+    KubeletConfig map[string]interface{}
+    // StatusCode is the result of the healthz endpoint for the node.
+    StatusCode int
+}
+
+// Metadata is all the stuff about the Sonobuoy run and how long it took to query the system.
+type Metadata struct {
+    // Config is the config used during this Sonobuoy run.
+    Config sonobuoy.Config
+    // QueryMetadata shows how long each object took to query.
+    QueryData []*sonobuoy.QueryData
+}
+
+// Plugin is a way to get at the files written by your plugin.
+type Plugin interface {
+    // Name returns the name of the plugin.
+    Name() string
+    // Walk iterates over every file generated by this plugin (everything under /plugins/{name}/) and executes the WalkFunc.
+    Walk(filepath.WalkFunc) error
+}
+
+// PodLog are all the logs gathered from an individual pod.
+type PodLog struct {
+    // Name is the name of the pod.
+    Name string
+    // Logs are the logs in io.Reader form because they can be quite large.
+    Logs io.Reader
+}
+```
+
+Some sample code using this interface:
+
+```go
+resultsClient := sonobuoy.NewResults("/path/to/tarball")
+
+resources := resultsClient.NamespacedResources()
+
+// The client could use it like this:
+// https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/helper.go#L155
+
+
+// Some information about the hosts configuration.
+hosts := tarballClient.Hosts()
+for _, host := range hosts {
+    fmt.Println("is the kubelet on", host.Name, "containerized?", host.KubeletConfig["containerized"])
+    if host.StatusCode!= "200" {
+        fmt.Println(host.Name, "is not ok according to the healthz endpoint")
+    }
+}
+
+// Metadata about the Sonobuoy run itself.
+metadata := tarballClient.Metadata()
+fmt.Println(metadata.Config()) // a sonobuoy.Config
+for _, queryData := range metadata.QueryData() {
+    fmt.Println(queryData) // a sonobuoy.queryData (TODO: Export.)
+}
+
+// Getting plugin files should be pretty generic to support all plugins.
+// Current assumptions include that a plugin will have some output.
+plugins := tarballClient.Plugins()
+walker := func(path string, info os.FileInfo, err error) error {
+    fmt.Println(path)
+
+}
+for _, plugin := range plugins {
+    fmt.Println("Found the following results for", plugin.Name())
+
+    // Read all the data from every file.
+    plugin.Walk(func walkFunc(filename string, info os.FileInfo, err error) error{
+        if err != nil {
+            return err
+        }
+        reader, ok := info.Sys().(io.Reader)
+        if !ok {
+            return errors.New("info.Sys() did not return an io.Reader!")
+        }
+        data, _ := ioutil.ReadAll(reader)
+        fmt.println(data)
+    })
+}
+
+podLogs := tarballClient.PodLogs()
```

Comment:
How do you associate all the pod data that you would need to access in order to determine the logs you want?